docs/root/intro/arch_overview/advanced/wasm.rst

```diff
@@ -0,0 +1,70 @@
+.. _arch_overview_wasm:
+
+Wasm
+====
+
+Envoy supports execution of the Wasm modules implemented against the `Proxy-Wasm specification <https://github.com/proxy-wasm/spec>`_.
+Currently, the `binary interface version 0.2.1 <https://github.com/proxy-wasm/spec/tree/main/abi-versions/v0.2.1>`_ is recommended.
+Wasm offers a portable and compact binary executable format that can be compiled once from a variety of languages
+(`C++ <https://github.com/proxy-wasm/proxy-wasm-cpp-sdk>`_, `Rust <https://github.com/proxy-wasm/proxy-wasm-rust-sdk>`_) and
+run anywhere. Please see the :ref:`sandbox example <install_sandboxes_wasm_filter>` for instructions on how to create and deploy
+a Wasm module.
+
```

Comment:
Suggest adding a short section on the different types of plugins, with links to their config protos:

```
Envoy provides several extension points at which Wasm plugins can be invoked:

* As an :ref:`HTTP filter <envoy.extensions.filters.http.wasm.v3.Wasm>`
* As a :ref:`network-level (L4) filter <envoy.extensions.filters.network.wasm.v3.Wasm>`
* As a :ref:`StatsSink <envoy.extensions.stat_sinks.wasm.v3.Wasm>`
* As an :ref:`AccessLogger <envoy.extensions.access_loggers.wasm.v3.WasmAccessLog>`
* As a :ref:`background service <envoy.extensions.wasm.v3.WasmService>`

The particular functions that Envoy invokes on a Wasm plugin depend on the
extension point at which it is configured.
```